### PR TITLE
ITL-6 Add sorting functionality

### PR DESCRIPTION
**Pull request by BiroAron**

### CODE REVIEW

File: src/components/header/ArrowButton.vue [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<template>
+  <button class="bg-black py-1 px-2 rounded-md font-semibold flex items-center">
+    <DownArrowIcon></DownArrowIcon>
+  </button>
+</template>
+
+<script setup lang="ts">
+//import UpArrowIcon from '../icons/UpArrowIcon.vue'
```
Comment (kincsokiss) [inline src/components/header/ArrowButton.vue]: Please remove the unused import.

File: src/components/header/Filter.vue [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      ></SortButton>
+      <SortButton
+        class="flex-grow p-2 bg-gray-300"
+        button-name="Priority"
+      ></SortButton>
+    </div>
+    <div
+      class="flex items-center desktop:justify-end phone:sutify-start phone:mt-2 w-full"
+    >
+      <ArrowButton class=""></ArrowButton>
```
Comment (kincsokiss) [inline src/components/header/Filter.vue]: Please remove the empty class attribute.

File: src/components/todo/ToDoList.vue [hunk 3]
```diff
@@ -4,6 +4,9 @@
       <Header @toggle-new-todo="toggleNewTodo"></Header>
 
       <Searchbar class="mb-15" @set-search-query="setSearcQuery" />
```
Comment (kincsokiss) [inline src/components/todo/ToDoList.vue]: There is a misspelling issue in `setSearcQuery`. It has to be `setSearchQuery`.

**Review by kincsokiss — CHANGES_REQUESTED** [review]

File: src/components/header/ArrowButton.vue [hunk 4]
```diff
@@ -0,0 +1,24 @@
+<template>
+  <button
+    class="bg-black py-2 px-3 rounded-md font-semibold flex items-center"
+    :class="setBgColor"
```
Comment (gerebdavid) [inline src/components/header/ArrowButton.vue]: This should be getBgColor

File: src/components/header/ArrowButton.vue [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  >
+    <DownArrowIcon v-if="sortAscending"></DownArrowIcon>
```
Comment (gerebdavid) [inline src/components/header/ArrowButton.vue:6]: Use v-if v-else

File: src/components/header/Filter.vue [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  sortAscending: boolean
+}
+const props = defineProps<Props>()
+
+const emit = defineEmits<{
+  (e: 'sortTodos', sortCriteria: string): void
+  (e: 'toggleSortOrder'): void
+}>()
+
+const activeortButtonColor = computed(() =>
```
Comment (gerebdavid) [inline src/components/header/Filter.vue]: There is a typo

File: src/components/header/Filter.vue [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    >
+      <ArrowButton
+        :sort-ascending="sortAscending"
+        @click="toggleSortOrder"
+      ></ArrowButton>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
```
Comment (gerebdavid) [inline src/components/header/Filter.vue:24]: Please follow the style guide, and move computeds before functions

File: src/components/todo/ToDoList.vue [hunk 8]
```diff
@@ -76,11 +69,78 @@ const filteredTodos = computed(() => {
...
+    : b.text.localeCompare(a.text)
+}
+
+function sortByDate(a: Todo, b: Todo) {
+  const dateA = new Date(a.date)
+  const dateB = new Date(b.date)
+
+  if (sortAscending.value) {
+    return dateA.getTime() - dateB.getTime()
+  } else {
```
Comment (gerebdavid) [inline src/components/todo/ToDoList.vue]: dont need else if you return inside if

File: src/components/todo/ToDoList.vue [hunk 9]
```diff
@@ -76,11 +69,78 @@ const filteredTodos = computed(() => {
...
+}
+
+function sortByPriority(a: Todo, b: Todo) {
+  const priorityOrder = ['Low', 'Medium', 'High']
+  const priorityA = priorityOrder.indexOf(a.priority)
+  const priorityB = priorityOrder.indexOf(b.priority)
+
+  if (sortAscending.value) {
+    return priorityA - priorityB
+  } else {
```
Comment (gerebdavid) [inline src/components/todo/ToDoList.vue]: same

**Review by kincsokiss — APPROVED** [review]

**Review by tamasm-wg — APPROVED** [review]

**Pull request merged**